pilot/pkg/networking/core/v1alpha3/tracing.go

```diff
@@ -136,21 +146,49 @@ func configureFromProviderConfig(pushCtx *model.PushContext, meta *model.NodeMet
 
 	switch provider := providerCfg.Provider.(type) {
 	case *meshconfig.MeshConfig_ExtensionProvider_Zipkin:
-		tracing, err = buildHCMTracing(pushCtx, providerCfg.Name, provider.Zipkin.Service, provider.Zipkin.Port, provider.Zipkin.MaxTagLength, zipkinConfigGen)
+		tracing, err = buildHCMTracing(pushCtx, envoyZipkin, provider.Zipkin.Service, provider.Zipkin.Port, provider.Zipkin.MaxTagLength, zipkinConfigGen)
 	case *meshconfig.MeshConfig_ExtensionProvider_Datadog:
-		tracing, err = buildHCMTracing(pushCtx, providerCfg.Name, provider.Datadog.Service, provider.Datadog.Port, provider.Datadog.MaxTagLength, datadogConfigGen)
+		tracing, err = buildHCMTracing(pushCtx, envoyDatadog, provider.Datadog.Service, provider.Datadog.Port, provider.Datadog.MaxTagLength, datadogConfigGen)
 	case *meshconfig.MeshConfig_ExtensionProvider_Lightstep:
-		tracing, err = buildHCMTracing(pushCtx, providerCfg.Name, provider.Lightstep.Service, provider.Lightstep.Port, provider.Lightstep.MaxTagLength,
-			func(hostname, clusterName string) (*anypb.Any, error) {
-				lc := &tracingcfg.LightstepConfig{
-					CollectorCluster: clusterName,
-					AccessTokenFile:  provider.Lightstep.AccessToken,
-				}
-				return protoconv.MessageToAnyWithError(lc)
-			})
+		// todo: read raw metadata and retrieve lightstep extensions (instead of relying on version)
+
+		// Envoy dropped support for Lightstep in v1.24+ (~istio 1.16+). So, we generate old-style configuration
+		// for lightstep for everything before 1.16, but OTel-based configuration for all other cases
+		useOTel := util.IsIstioVersionGE116(model.ParseIstioVersion(meta.IstioVersion))
+		if useOTel {
+			tracing, err = buildHCMTracing(pushCtx, envoyOpenTelemetry, provider.Lightstep.Service, provider.Lightstep.Port, provider.Lightstep.MaxTagLength,
+				func(hostname, clusterName string) (*anypb.Any, error) {
```

Comment:
nit: To be consistent, abstract a function as other provider

could be in a follow up